app/src/ui/lib/text-box.tsx

```diff
@@ -177,6 +177,27 @@ export class TextBox extends React.Component<ITextBoxProps, ITextBoxState> {
     )
   }
 
+  private onKeyDown = (event: React.KeyboardEvent<HTMLInputElement>) => {
+    if (
+      this.state.value !== '' &&
+      this.props.type === 'search' &&
+      event.key === 'Escape'
+    ) {
+      const value = ''
+
+      event.preventDefault()
+      this.setState({ value })
+
+      if (this.props.onValueChanged) {
+        this.props.onValueChanged(value)
+      }
+    }
+
+    if (this.props.onKeyDown != null) {
```

Comment:
Could we make this `(this.props.onKeyDown !== undefined)` since it's typed as undefined or function or just `if(this.props.onKeyDown)`?